### PR TITLE
Spooler bug

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[b042f8af7586ed789a17c9e7b8ab42f255eebcf7] [ERROR] Errors: 
[b042f8af7586ed789a17c9e7b8ab42f255eebcf7] [ERROR]   MailsShouldBeWellReceived.hugeLoadShouldBeProcessedEntirely:95 » ConditionTimeout
[b042f8af7586ed789a17c9e7b8ab42f255eebcf7] [ERROR]   MailsShouldBeWellReceived.hugeLoadShouldBeProcessedEntirely:95 » ConditionTimeout
[b042f8af7586ed789a17c9e7b8ab42f255eebcf7] [ERROR]   MailsShouldBeWellReceived.hugeLoadShouldBeProcessedEntirely:95 » ConditionTimeout
[b042f8af7586ed789a17c9e7b8ab42f255eebcf7] [ERROR]   MailsShouldBeWellReceived.hugeLoadShouldBeProcessedEntirely:95 » ConditionTimeout
[b042f8af7586ed789a17c9e7b8ab42f255eebcf7] [ERROR]   MailsShouldBeWellReceived.hugeLoadShouldBeProcessedEntirely:95 » ConditionTimeout
```

**Comment by blackheaven** [comment]
test this please

**Comment by remk** [comment]
Cool, could you just improve the commit messages ?

File: server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/JamesMailSpoolerTest.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        spooler.setMailQueueFactory(queueFactory);
+
+        MailProcessor mailProcessor = mock(MailProcessor.class);
+        AtomicBoolean hasFailed = new AtomicBoolean(false);
+        doAnswer(ignored -> {
+            if (!hasFailed.getAndSet(true)) {
+                throw new RuntimeException("Arbitrary failure");
+            }
+            return null;
+        }).when(mailProcessor).service(any());
```
Comment (rouazana) [inline server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/JamesMailSpoolerTest.java]: you don't need a if, you can write it directly with mockito: https://stackoverflow.com/questions/11785498/simulate-first-call-fails-second-call-succeeds/41272746#41272746

File: server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/JamesMailSpoolerTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        spooler.setMailQueueFactory(queueFactory);
+
+        MailProcessor mailProcessor = mock(MailProcessor.class);
+        AtomicBoolean hasFailed = new AtomicBoolean(false);
+        doAnswer(ignored -> {
+            if (!hasFailed.getAndSet(true)) {
+                Thread.currentThread().interrupt();
+            }
+            return null;
+        }).when(mailProcessor).service(any());
```
Comment (rouazana) [inline server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/JamesMailSpoolerTest.java]: idem

**Comment by blackheaven** [comment]
done

**Comment by Arsnael** [comment]
Build doesn't exist anymore...

test this please

**Pull request merged**